### PR TITLE
Add negative paddings support for pad

### PR DESCRIPTION
**Pull request by chinhuang007**
This PR is to add negative paddings support for pad operator
according to ONNX spec, negative pads mean to remove elements.

It also should address the issue
https://github.com/onnx/onnx-tensorflow/issues/584

The current assumption is that paddings can either positive or
negative, not mix of them, since ONNX spec does not define
such scenario and behavior.

### CODE REVIEW

**Comment by tjingrant** [comment]
Hi Chin, how does this approach compare with, say using `tf.slice`? It seems like you have to materialize the coordinate vector for each pixel that is not removed by negative padding, and using `tf.gather` also carries performance penalties, given that it's fundamentally a sparse operation.

**Comment by chinhuang007** [comment]
Thanks for the review, Tian. The logic is similar to tf.slice. However getting the proper begins and sizes out of the 1-d tensor pads and input shape requires calculation (including gather I believe) as well. Would you say using tf.slice will definitely improve performance? I could take a look at the alternative.

**Comment by chinhuang007** [comment]
@tjingrant I changed the code to use tf.slice. It is much simpler. Thanks very much for your suggestion. Please review.

File: onnx_tf/handlers/backend/pad.py [hunk 1]
```diff
@@ -17,40 +16,63 @@ def _common(cls, node, **kwargs):
...
+        return i + 1, x
+
+      # tf requires int32 paddings
+      paddings = tf.cast(tf.transpose(tf.reshape(paddings, [2, num_dim])),
+                         dtype=tf.int32)
+
+      if mode.lower() == "edge":
+        paddings = tf.reshape(paddings, [-1])
+        max_i = tf.reduce_max(paddings)
+        _, x = tf.while_loop(
```
Comment (tjingrant) [inline onnx_tf/handlers/backend/pad.py:53]: Hi Chin, I either forgot or never understood what this part is doing, could you give me some hints about what's going on with the while loop? Some comments may help with future reviews, too.

Comment (chinhuang007) [reply onnx_tf/handlers/backend/pad.py:53]: The while loop code is not changed in this PR, just added with indentation. It does look a bit strange. I will need to investigate what it is doing and whether it works as expected.

Comment (chinhuang007) [reply onnx_tf/handlers/backend/pad.py:57]: Okay, now I understand the while loop is to replace the python implementation of edge mode, np.pad(x, pads, mode="edge"). Basically it goes through the tf.pad 'SYMMETRIC' mode to pad one value at a time for both sides and all dimensions. I added some comments and hope it helps.
